servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpSerializers.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright © 2021 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.servicetalk.http.api;
+
+import io.servicetalk.serializer.api.SerializerDeserializer;
+import io.servicetalk.serializer.api.StreamingSerializerDeserializer;
+import io.servicetalk.serializer.utils.FixedLengthStreamingSerializer;
+import io.servicetalk.serializer.utils.VarIntLengthStreamingSerializer;
+
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import static io.servicetalk.buffer.api.CharSequences.newAsciiString;
+import static io.servicetalk.http.api.HeaderUtils.hasContentType;
+import static io.servicetalk.http.api.HttpHeaderNames.CONTENT_TYPE;
+import static io.servicetalk.http.api.HttpHeaderValues.APPLICATION_JSON;
+import static io.servicetalk.http.api.HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED;
+import static io.servicetalk.http.api.HttpHeaderValues.APPLICATION_X_WWW_FORM_URLENCODED_UTF_8;
+import static io.servicetalk.http.api.HttpHeaderValues.TEXT_PLAIN;
+import static io.servicetalk.http.api.HttpHeaderValues.TEXT_PLAIN_US_ASCII;
+import static io.servicetalk.http.api.HttpHeaderValues.TEXT_PLAIN_UTF_8;
+import static io.servicetalk.serializer.utils.StringSerializer.stringSerializer;
+import static java.nio.charset.StandardCharsets.US_ASCII;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * Factory for creation of {@link HttpSerializerDeserializer} and {@link HttpStreamingSerializerDeserializer}.
+ */
+public final class HttpSerializers {
+    private static final CharSequence APPLICATION_TEXT_FIXED = newAsciiString("application/text-fix-int");
```

Comment:
I think `application` I more appropriate bcz it isn't strictly human readable (binary data for integer encoding).

https://www.rfc-editor.org/rfc/rfc2046#section-3
> text -- textual information. ... In
          particular, formats that employ embeddded binary
          formatting information are not considered directly
          readable.

> application -- some other kind of data, typically
          either uninterpreted binary data or information to be
          processed by an application. ... and word
          processing formats that are not directly readable.

https://www.rfc-editor.org/rfc/rfc2046#section-4.1
> The "text" media type is intended for sending material which is
   principally textual in form.

https://datatracker.ietf.org/doc/html/rfc3023#section-3
> If an XML document -- that is, the unprocessed, source XML document
   -- is readable by casual users, text/xml is preferable to
   application/xml.

Reply:
Ok, because we have a length prefix, it's not 100% text data. I would agree, `application/...` is more appropriate. Users can configure their own header value, if necessary. Thanks for the links!